From d9776e3f9f98fe1d8a3cc72b398e324d89e10529 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 3 Jun 2025 13:37:39 +0100
Subject: [PATCH] manylinux-2.28: do not try to find non-oldglibc header

libstdc++ headers attempt to locate and include single_threaded.h
which is present on the system, but for the system-wide / higher glibc
version.

Instead do not try to dynamically include and use that symbol, as it
only exists in glibc 2.32+.
---
 libstdc++-v3/include/ext/atomicity.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/libstdc++-v3/include/ext/atomicity.h b/libstdc++-v3/include/ext/atomicity.h
index 98f745c4f70..f7fea75d3b7 100644
--- a/libstdc++-v3/include/ext/atomicity.h
+++ b/libstdc++-v3/include/ext/atomicity.h
@@ -36,9 +36,6 @@
 #include <bits/c++config.h>
 #include <bits/gthr.h>
 #include <bits/atomic_word.h>
-#if __has_include(<sys/single_threaded.h>)
-# include <sys/single_threaded.h>
-#endif
 
 namespace __gnu_cxx _GLIBCXX_VISIBILITY(default)
 {
@@ -50,8 +47,6 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
   {
 #ifndef __GTHREADS
     return true;
-#elif __has_include(<sys/single_threaded.h>)
-    return ::__libc_single_threaded;
 #else
     return !__gthread_active_p();
 #endif
-- 
2.48.1

